templates/workspace_services/azureml/template_schema.json

```diff
@@ -1,11 +1,386 @@
+                  ]
+                },
+                {
+                  "name": "AzureML_Dependancies",
+                  "description": "AzureML Dependancies",
+                  "source_addresses": [
+                    "{{ resource.properties.workspace_services_subnet_address_prefix }}"
+                  ],
+                  "destination_addresses": [
+                    "AzureActiveDirectory",
+                    "AzureResourceManager",
+                    "MicrosoftContainerRegistry"
+                  ],
+                  "destination_ports": [
+                    "443"
+                  ],
+                  "protocols": [
+                    "TCP"
+                  ]
+                },
+                {
+                  "name": "AzureML_Storage",
+                  "description": "AzureML Storage",
+                  "source_addresses": [
+                    "{{ resource.properties.workspace_services_subnet_address_prefix }}"
+                  ],
+                  "destination_addresses": [
+                    "{{ resource.properties.storage_tag }}"
+                  ],
+                  "destination_ports": [
+                    "443",
+                    "445"
+                  ],
+                  "protocols": [
+                    "TCP"
+                  ]
+                }
+              ]
+            }
+          },
+          {
+            "name": "rule_collections",
+            "type": "array",
+            "arraySubstitutionAction": "replace",
+            "arrayMatchField": "name",
+            "value": {
+              "name": "arc_svc_{{ resource.id }}_azureml",
+              "action": "Allow",
+              "rules": [
+                {
+                  "name": "AzureML",
+                  "description": "AzureML rules",
+                  "source_addresses": [
+                    "{{ resource.properties.workspace_services_subnet_address_prefix }}"
+                  ],
+                  "target_fqdns": [
+                    "aadcdn.msftauth.net",
+                    "ml.azure.com"
+                  ],
+                  "protocols": [
+                    {
+                      "port": "443",
+                      "type": "Https"
+                    }
+                  ]
+                }
+              ]
+            }
+          }
+        ]
+      }
+    ],
+    "uninstall": [
+      {
+        "stepId": "260421b3-7308-491f-b531-e007cdc0ff48",
+        "stepTitle": "Add network firewall rules for azureml",
+        "resourceTemplateName": "tre-shared-service-firewall",
+        "resourceType": "shared-service",
+        "resourceAction": "upgrade",
+        "properties": [
+          {
+            "name": "network_rule_collections",
+            "type": "array",
+            "arraySubstitutionAction": "remove",
+            "arrayMatchField": "name",
+            "value": {
+              "name": "nrc_svc_{{ resource.id }}_azureml",
+              "action": "Allow",
```

Comment:
can ditch everything other than `name` in here as we're removing the item from the array

Reply:
Done, that look ok?